NewForce-at-Mountwest/bangazon-site-the-grave-trolls
# User's can Add and View their Payment Types

# Description

User's can add a payment type and view all payment types associated with their account

# Related Tickets

https://trello.com/c/5uQylEuS/7-7-users-can-add-payment-type
https://trello.com/c/U1fuH0w8/23-75-users-can-view-all-payment-types

## Type of change


# Testing Instructions

* Add/commit your current changes to your working branch

* Run **git checkout master**

* Run **git fetch --all**

* Run **git checkout damAddPaymentType**

* If you don't already have it create `AppSettings.json` (in BangazonSite project) or edit it with the following code:
`{
  "Logging": {
    "LogLevel": {
      "Default": "Warning"
    }
  },
  "AllowedHosts": "*",
  "ConnectionStrings": {
    "DefaultConnection": "Server=localhost\\SQLEXPRESS;Database=BangazonSite;Trusted_Connection=True;"
  }
}`

* The `"DefaultConnection"` will be different for some people make sure that it references the SQL server running on your machine

- If you DON'T have seed data or the initial database, open Package Manager Console and run `Add-Migration` and `Update-Database` to add these.

* In Visual Studio click the green play button at the top that says **IIS Express** or **F5** to start the browser

* Your browser should load a url in the address bar that looks like this **https://localhost:44345** 

* If your not logged in and display a page that says **Welcome** and the following affordances:
     *Home
     *Register 
     *Login

* If don't have an account click on the **Register** affordance to create one otherwise just log in

* Once logged in the top right coner of the screen you should see something like **Hello email@youremailaddress.com**

* Click on your email address and this will take you to a screen to manage your account settings

* Click on **Payment Options** located on the left side of your screen

* If you already have payment types associated with your user you will directed to a screen displaying all of the user's payment types with an affordance to **Create New**

* If you don't have any payment types associated witth your user you will be directed to a screen that says **No Types of Payment Associated With This Account Yet** with an affordance to **Add A New Payment Type?**

* Click on either of the two affordances to be directed to a form to fill out information about your payment type

* The **Description** and **AccountNumber** inputs are required if left blank they will tell you that they are required 
**Note:** Not checking the active box will defualt to your payment type being inactive

* Once you input all the information correctly and click the create affordance you will be redirected back to a view of your payment types and you should see the payment type you just created


# Checklist:
